src/Compilers/CSharp/Test/Emit/PDB/PDBLocalFunctionTests.cs

```diff
@@ -0,0 +1,450 @@
+﻿using Microsoft.CodeAnalysis.CSharp.Test.Utilities;
+using Microsoft.CodeAnalysis.Test.Utilities;
+using Xunit;
+
+namespace Microsoft.CodeAnalysis.CSharp.UnitTests.PDB
+{
+    public class PDBLocalFunctionTests : CSharpPDBTestBase
+    {
+        [Fact]
+        public void ClosuresInCtor()
+        {
+            var source = @"
+using System;
+
+class B
+{
+    public B(Func<int> f) { }
+}
+
+class C : B
+{
+    int r;
+
+    public C(int a, int b) : base(() => a) 
+    {
+        int c = 1;
+        int f() => b;
+        int g() => f();
+        int h() => c;
+        r = g() + h();
+    }
+}
+";
+
+            var c = CreateCompilationWithMscorlibAndSystemCore(source, options: TestOptions.DebugDll);
+            c.VerifyDiagnostics();
+
+            c.VerifyPdb(@"
+<symbols>
+  <methods>
+    <method containingType=""B"" name="".ctor"" parameterNames=""f"">
+      <customDebugInfo>
+        <using>
+          <namespace usingCount=""1"" />
+        </using>
+      </customDebugInfo>
+      <sequencePoints>
+        <entry offset=""0x0"" startLine=""6"" startColumn=""5"" endLine=""6"" endColumn=""26"" />
+        <entry offset=""0x7"" startLine=""6"" startColumn=""27"" endLine=""6"" endColumn=""28"" />
+        <entry offset=""0x8"" startLine=""6"" startColumn=""29"" endLine=""6"" endColumn=""30"" />
+      </sequencePoints>
+      <scope startOffset=""0x0"" endOffset=""0x9"">
+        <namespace name=""System"" />
+      </scope>
+    </method>
+    <method containingType=""C"" name="".ctor"" parameterNames=""a, b"">
+      <customDebugInfo>
+        <forward declaringType=""B"" methodName="".ctor"" parameterNames=""f"" />
+        <encLocalSlotMap>
+          <slot kind=""30"" offset=""-1"" />
+          <slot kind=""30"" offset=""0"" />
+        </encLocalSlotMap>
+        <encLambdaMap>
+          <methodOrdinal>1</methodOrdinal>
+          <closure offset=""-1"" />
+          <closure offset=""0"" />
+          <lambda offset=""-2"" closure=""0"" />
+          <lambda offset=""42"" closure=""0"" />
+          <lambda offset=""65"" closure=""0"" />
+          <lambda offset=""90"" />
```

Comment:
><lambda offset=""90"" /> [](start = 10, length = 24)

This doesn't look right. The local function h should be assigned closure #1. See the same test for lambdas.

Reply:
Check other test cases to see if there are any differences between lambdas and local function. If there are then we need to know why they are different.


---
In reply to: [137146584](https://github.com/dotnet/roslyn/pull/21777#discussion_r137146584) [](ancestors = 137146584)

Reply:
I could not find any other essential instances.

---
In reply to: [137146870](https://github.com/dotnet/roslyn/pull/21777#discussion_r137146870) [](ancestors = 137146870,137146584)

Reply:
As far as I can see, the closure is not set for the last local function within [SynthesizeLoweredFunctionMethods](http://source.roslyn.io/#Microsoft.CodeAnalysis.CSharp/Lowering/LambdaRewriter/LambdaRewriter.cs,f2141dfd2141522c). This happens because ContainingEnvironmentOpt is null for it. The ContainingEnvironmentOpt is set in [ComputeLambdaScopesAndFrameCaptures](http://source.roslyn.io/#Microsoft.CodeAnalysis.CSharp/Lowering/LambdaRewriter/LambdaRewriter.Analysis.cs,155). The one for the last local function is not set because it is a struct. Please note that all other local functions in the test are not treated as structs.

@agocke, could you please help with this? What would be the right approach to get the right closure for the last local function?

Reply:
@ivanbasov If struct closures are not recorded properly we may have to find a different serialization mechanism for the PDB. What would you do if a local function captures multiple struct closures?

Reply:
We determined that this is actually expected. Let's add more cases like this to EditAndContinueTests in next PR

---
In reply to: [137160824](https://github.com/dotnet/roslyn/pull/21777#discussion_r137160824) [](ancestors = 137160824,137146870,137146584)